OctoKit/Issue.swift

```diff
@@ -3,34 +3,80 @@ import RequestKit
 
 // MARK: model
 
+public enum IssueState: String {
+    case Open = "open"
+    case Closed = "closed"
+}
+
 @objc public class Issue: NSObject {
-    public let id: Int
-    public var url: String?
+    public var id: Int
+    public var url: NSURL?
+    public var repositoryURL: NSURL?
+    public var labelsURL: NSURL?
+    public var commentsURL: NSURL?
+    public var eventsURL: NSURL?
+    public var htmlURL: NSURL?
     public var number: Int?
+    public var state: IssueState?
     public var title: String?
     public var body: String?
-    public var state: String?
     public var user: User?
+    public var labels: [Label]?
     public var assignee: User?
-    public var createdAt: NSDate?
-    public var closedAt: NSDate?
-    public var updatedAt: NSDate?
+    public var milestone: Milestone?
+    public var locked: Bool?
+    public var numberOfComments: Int?
+    public var closureDate: NSDate?
+    public var creationDate: NSDate?
+    public var updateDate: NSDate?
+    public var closer: User?
     
     public init(_ json: [String: AnyObject]) {
         if let id = json["id"] as? Int {
             self.id = id
-            url = json["url"] as? String
+            if let urlString = json["url"] as? String, url = NSURL(string: urlString) {
+                self.url = url
+            }
+            if let urlString = json["repository_url"] as? String, url = NSURL(string: urlString) {
+                repositoryURL = url
+            }
+            if let urlString = json["labels_url"] as? String, url = NSURL(string: urlString) {
+                labelsURL = url
+            }
+            if let urlString = json["comments_url"] as? String, url = NSURL(string: urlString) {
+                commentsURL = url
+            }
+            if let urlString = json["events_url"] as? String, url = NSURL(string: urlString) {
+                eventsURL = url
+            }
+            if let urlString = json["html_url"] as? String, url = NSURL(string: urlString) {
+                htmlURL = url
+            }
             number = json["number"] as? Int
+            switch json["state"] as? String ?? "" {
```

Comment:
It would make sense to initialise the `enum` with the `rawValue` instead of replicating the logic.
```swift
state = IssueState(rawValue: json["state"] as? String ?? "")
```